std/variant.d

```diff
@@ -339,7 +339,19 @@ private:
             static if (target.size < A.sizeof)
             {
                 if (target.type.tsize < A.sizeof)
-                    *cast(A**)&target.store = new A;
+                {
+                    static if (is(A a == U[n], U, size_t n))
+                    {
+                        pragma(msg, "A");
+                        A* p = cast(A*)(new U[n]).ptr;
```

Comment:
Correct me if I'm wrong, but if `A == U[n]` then `typeof((new U[n]).ptr)` is already `A*` and thus there's no need for this cast.